R/helpers_data.R

```diff
@@ -0,0 +1,235 @@
+#' @param ... further arguments passed to `data.frame` constructor.
+#'   It can be e.g. an extra `column_name = value` pair based on a slot
+#'   from `x` (which in this case might be a subclass of `Data`)
+#'   which does not appear in `Data`.
+#' @return A [`data.frame`] object with columns patient, ID, cohort, dose and
+#' toxicity.
+#' @describeIn h_plot_data_df Class specific method for [`Data`]
+setMethod(
+  f = "h_plot_data_df",
+  signature = signature(data = "Data"),
+  definition = function(data, blind = FALSE, legend = TRUE, ...) {
+    df <- data.frame(
+      patient = seq_along(data@x),
+      ID = paste(" ", data@ID),
+      cohort = data@cohort,
+      dose = data@x,
+      toxicity = ifelse(data@y == 1, "Yes", "No"),
+      ...
+    )
+    df <- h_blind_plot_data(df, blind, data@placebo, data@doseGrid[1])
+    df
+  }
+)
+
+# DataOrdinal ----
+
+#' Helper Function for the Plot Method of [`DataOrdinal`]
+#'
+#' @param data (`DataOrdinal`)\cr object from which data is extracted and converted
+#'   into a data frame.
+#' @param blind (`flag`)\cr should data be blinded?
+#'   If `TRUE`, then for each cohort, all DLTs are assigned to the first
+#'   subjects in the cohort. In addition, the placebo (if any) is set to the
+#'   active dose level for that cohort.
+#' @param ... further arguments passed to `data.frame` constructor.
+#'   It can be e.g. an extra `column_name = value` pair based on a slot
+#'   from `x` (which in this case might be a subclass of `Data`)
+#'   which does not appear in `Data`.
+#' @return A [`data.frame`] object with values to plot.
+#' @describeIn h_plot_data_df Class specific method for [`DataOrdinal`]
+setMethod(
+  f = "h_plot_data_df",
+  signature = signature(data = "DataOrdinal"),
+  definition = function(data, blind = FALSE, legend = TRUE, ...) {
+    df <- data.frame(
+      patient = seq_along(data@x),
+      ID = paste(" ", data@ID),
+      cohort = data@cohort,
+      dose = data@x,
+      toxicity = names(data@yCategories)[1 + data@y],
+      ...
+    )
+    df <- h_blind_plot_data(df, blind, data@placebo, data@doseGrid[1])
+    df
+  }
+)
+
+
+# h_plot_data_dataordinal
+
+## Data ----
+
+#' Helper Function for the Plot Method of the Data and DataOrdinal Classes
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' A method that creates a plot for [`Data`]  and [`DataOrdinal`] objects.
+#'
+#' @param x (`Data`)\cr object we want to plot.
+#' @param y (`missing`)\cr missing object, for compatibility with the generic
+#'   function.
+#' @param blind (`flag`)\cr indicates whether to blind the data.
+#'   If `TRUE`, then placebo subjects are reported at the same level
+#'   as the active dose level in the corresponding cohort,
+#'   and DLTs are always assigned to the first subjects in a cohort.
+#' @param tox_labels (`character`)\cr the toxicity category labels
+#' @param tox_shapes (`integer`)\cr the shapes used to plot the various toxicity
+#' categories
```

Comment:
this is repeated from the plot method I believe, can we inherit arg description from there?

Reply:
I believe this is not an inherited method.  It is a function. Though it is called by two inherit methods - `plot-Data` and `plot-DataOrdinal` - and does the majority of the donkey work, particularly for `plot_Data`.

Reply:
Yeah I did not mean this is an inherited method, just that the argument description seems duplicate. Can we inherit the argument descriptions with `@inheritParam` ?

Reply:
Ah.  OK.  Now I understand.  In fact, `@inheritParams` isn't just not necessary, it generates an error.  The `@rdname plot-Data` seems to manage that automatically.  I have also modified `plot-Data` / `plot-DataOrdinal`.